securedrop_client/queue.py

```diff
@@ -41,12 +41,20 @@ def _process(self, exit_loop: bool) -> None:
                 job._do_call_api(self.api_client, session)
             except RequestTimeoutError:
                 self.last_job = job  # "cache" the last job since we can't re-queue it
+                session.close()
+                return
+            except ApiInaccessibleError:
+                # This is a guard against #397, we should pause the queue execution when this
+                # happens in the future and flag the situation to the user (see ticket #379).
+                logger.error('Client is not authenticated, skipping job...')
+                session.close()
                 return
 
             # process events to allow this thread to handle incoming signals
             QApplication.processEvents()
```

Comment:
[nit] you _could_ throw this into the try block and then remove all the `return`s + makes it more readable

Reply:
After giving this another look, I see that returning upon exception is what the code was doing previously (for `RequestTimeoutError`) so it makes sense to just return for `ApiInaccessibleError`s. 

Code for retries, pausing, and clearing the queue is out of the scope here and still needs to be implemented. So this probably shouldn't be refactored. I made the "nit" comment a little too quickly without realizing that it would involve a lot more changes around queue behavior, some of which we already have Issues for. 

But I like your idea for writing more tests around the queue. I think it would be find to do it outside of this PR, but either way.

Reply:
you were making a good point here so I've done this prior to merge in a `finally` block for readability

since we don't have the pause behavior yet (for #379), I thought I'd write tests to at least ensure that the queue does not die or the app crash (as described in #379) when timeouts, auth errors, or other unexpected exceptions occur, as well as ensure that queue threads are restarted if they terminate for robustness (in the diff here I chose to do this when each job is enqueued - when we add queue persistence the queue should be picking up where it left off). This means the PR also closes #380 which I had previously partly implemented in this PR so that I could log in, log out, and log back in and use the same queue